src/LiveComponent/assets/dist/Component/index.d.ts

```diff
@@ -30,8 +33,34 @@ export default class Component {
     addPlugin(plugin: PluginInterface): void;
     connect(): void;
     disconnect(): void;
-    on(hookName: string, callback: (...args: any[]) => void): void;
-    off(hookName: string, callback: (...args: any[]) => void): void;
+    on(hookName: 'connect', callback: (component: Component) => MaybePromise): void;
```

Comment:
Instead of having all of those definitions why not just use type ComponentHookName?
on(hookName: ComponentHookName, callback: (component: Component) => MaybePromise): void;

Reply:
Because the callbacks receive different arguments depending on the event. By explicitly overloading this method, TypeScript (and your IDE) now knows which events are available, and the arguments that they each receive. It's a much nicer developer experience, and it should work for any IDE even when not using TypeScript.